tests/test_pf7.py

```diff
@@ -0,0 +1,433 @@
+            actual_calldata_dictionary,
+        ) = self.test_pf7_class._subset_extended_dictionary(
+            ["DP", "GQ", "AC", "AF", "ANN_Feature_Type"]
+        )
+        expected_calldata_dictionary = {
+            "DP": [DIM_VARIANT, DIM_SAMPLE],
+            "GQ": [DIM_VARIANT, DIM_SAMPLE],
+        }
+        expected_variants_dictionary = {
+            "DP": [DIM_VARIANT],
+            "AC": [DIM_VARIANT, DIM_ALT_ALLELE],
+            "AF": [DIM_VARIANT, DIM_ALT_ALLELE],
+            "ANN_Feature_Type": [DIM_VARIANT, DIM_ALT_ALLELE],
+        }
+        self.assertEqual(actual_variants_dictionary, expected_variants_dictionary)
+        self.assertEqual(actual_calldata_dictionary, expected_calldata_dictionary)
+
+    def test_subset_extended_dictionary_raises_error(self):
+        with self.assertRaises(ValueError):
+            self.test_pf7_class._subset_extended_dictionary(
+                ["DP", "AC", "BAD_VARIABLE"]
+            )
+
+    def test_add_coordinates(self):
+        var_names_for_outputs = {
+            "POS": "position",
+            "CHROM": "chrom",
+            "FILTER_PASS": "filter_pass",
+        }
+        actual_coordinates = self.test_pf7_class._add_coordinates(
+            self.test_zarr_root, True, "native", var_names_for_outputs
+        )
+        self.assertEqual(
+            list(actual_coordinates.keys()),
+            ["variant_position", "variant_chrom", "sample_id"],
+        )
+
+        values = [actual_coordinates[k] for k in actual_coordinates]
+        for value in values:
+            assert isinstance(value[1], da.Array)
+        dimensions = [i[0] for i in values]
+        self.assertEqual(
+            dimensions,
+            [["variants"], ["variants"], ["samples"]],
+        )
+
+    def test_add_data_vars(self):
+        var_names_for_outputs = {
+            "POS": "position",
+            "CHROM": "chrom",
+            "FILTER_PASS": "filter_pass",
+        }
+        actual_vars = self.test_pf7_class.add_data_vars(
+            self.test_zarr_root, True, "native", var_names_for_outputs
+        )
+        self.assertEqual(
+            list(actual_vars.keys()),
+            [
+                "variant_allele",
+                "variant_filter_pass",
+                "variant_is_snp",
+                "variant_numalt",
+                "variant_CDS",
+                "call_genotype",
+                "call_AD",
+            ],
+        )
+        values = [actual_vars[k] for k in actual_vars]
+        for value in values:
+            assert isinstance(value[1], da.Array)
+        dimensions = [i[0] for i in values]
+        self.assertEqual(
+            dimensions,
+            [
+                ["variants", "alleles"],
+                ["variants"],
+                ["variants"],
+                ["variants"],
+                ["variants"],
+                ["variants", "samples", "ploidy"],
+                ["variants", "samples", "alleles"],
+            ],
+        )
```

Comment:
This tests (what should be) a private method, see comment above re testing private methods.

Reply:
See comment above about testing this private method